### PR TITLE
Issue 75

### PR DESCRIPTION
**Pull request by MichaelPHanks**
Season endpoints, with preloading for leagues in the season done

### CODE REVIEW

File: backend/internal/db/migrations/migrate.go
Comment (Skylude) [inline backend/internal/db/migrations/migrate.go]: ❌ rather than add it here, add it to the auto migrations below and use the `make nuke-migrations` command to reset the database. If you do keep it here you have to ensure you check on both the create and the drop in rollback that the table doesn't exist on the migration (because it could if it's an existing database) and that the table exists on the drop (because it might not)

File: backend/internal/server/apis/groups/seasons.go
Comment (Skylude) [inline backend/internal/server/apis/groups/seasons.go]: ⚠️ I don't think we need to log the request each time -- even though it's debug

File: backend/internal/server/apis/groups/seasons.go
Comment (Skylude) [inline backend/internal/server/apis/groups/seasons.go]: ℹ️ unnecessary comment

File: backend/internal/server/apis/groups/seasons.go
Comment (Skylude) [inline backend/internal/server/apis/groups/seasons.go]: ⚠️ unnecessary comment

File: backend/internal/server/apis/groups/seasons.go
Comment (Skylude) [inline backend/internal/server/apis/groups/seasons.go]: ℹ️ change log to "Could not save season to database"

File: backend/internal/server/server.go
Comment (Skylude) [inline backend/internal/server/server.go:19]: ℹ️ not sure on the use of "groups" here.

File: static/oas/v1/season/season.yml
Comment (Skylude) [inline static/oas/v1/season/season.yml:1]: These should all begin with a / to follow openapi spec.

File: static/oas/v1/season/season.yml
Comment (Skylude) [inline static/oas/v1/season/season.yml]: ℹ️ The schema should be defined so you shouldn't have to re-list all of the parameters. Instead reference the Season schema and if there are any caveats / things to note -- potentially mention this endpoint does not create schedules, leagues or registrations.

File: static/oas/v1/season/season.yml
Comment (Skylude) [inline static/oas/v1/season/season.yml]: ❗ I actually don't think we want to be able to create registrations when we are creating a season because we haven't yet added players etc. So this will always start as an empty []. Same for schedule and leagues. Potentially we need a way to do this via CSV when bulk uploading or copying but not from the UI.

**Review by Skylude — CHANGES_REQUESTED** [review]
Left a few comments:
ℹ️ - Suggestions
⚠️ - Stronger Suggestions that I think you should likely change
❗ ❌  - Required changes before my approval.

Overall looks great just a few comments!

File: backend/internal/db/migrations/migrate.go
Comment (MichaelPHanks) [reply backend/internal/db/migrations/migrate.go]: Fixed

File: backend/internal/server/apis/groups/seasons.go
Comment (MichaelPHanks) [reply backend/internal/server/apis/groups/seasons.go]: Agreed, will fix

File: backend/internal/server/apis/groups/seasons.go
Comment (MichaelPHanks) [reply backend/internal/server/apis/groups/seasons.go]: Removed

File: backend/internal/server/apis/groups/seasons.go
Comment (MichaelPHanks) [reply backend/internal/server/apis/groups/seasons.go]: Removed

File: backend/internal/server/apis/groups/seasons.go
Comment (MichaelPHanks) [reply backend/internal/server/apis/groups/seasons.go]: Agreed, just fixed

File: backend/internal/server/server.go
Comment (MichaelPHanks) [reply backend/internal/server/server.go:19]: I assume you saw we changed the file structure for this, but in case not we are putting seasons, leagues, teams, games, etc. into a folder so we don't have too many separate folders and they seem to fit nicely in this way

File: static/oas/v1/season/season.yml
Comment (MichaelPHanks) [reply static/oas/v1/season/season.yml:1]: I attempted to add / to seasons at the beginning and no matter what I did it wouldn't work for me

Comment (MichaelPHanks) [reply static/oas/v1/season/season.yml:1]: The file just wouldn't load for the api docs

File: static/oas/v1/season/season.yml
Comment (MichaelPHanks) [reply static/oas/v1/season/season.yml]: Fixed

File: static/oas/v1/season/season.yml
Comment (MichaelPHanks) [reply static/oas/v1/season/season.yml]: Ah I see, I just figured that I would add a league for now so I could test preloading the leagues for the season. I'll remove all of these considering they shouldn't even be created yet.

**Review by Skylude — APPROVED** [review]

**Pull request merged**
